### PR TITLE
Cleanup of ClusterServiceImpl

### PR DESCRIPTION
**Pull request by Donnerbart**
This PR will be a bit more fun after the very simple cleanups before. I mostly fixed simple CheckStyle issues like magic numbers, 130 line width, trailing comments etc. I also aligned the format of comments and log messages. After that I took care about NPath and Cyclomatic complexity and pulled out some methods or adjusted the code flow to get rid of those issues.

I have quite high confidence that I didn't break anything. A lot of changes were done via IDEA refactoring tools. I ran the cluster tests 3 times including code coverage. There where no failures with 84% line coverage. The biggest untested part is the `ping()` method, which I didn't change significantly. The remaining untested parts are mostly the `toString()` method and logging branches (see https://hazelcast-l337.sonar.cloudbees.com/resource/index/11063?display_title=true&metric=coverage).

I hope I didn't do any controversial things. There maybe things which we can discuss, but I hope we can agree that the class is in better shape than before with this PR.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java [hunk 1]
```diff
@@ -919,28 +952,6 @@ public void merge(final Address newTargetAddress) {
         }
     }
 
-    private <V> V waitOnFutureInterruptible(Future<V> future, long timeout, TimeUnit timeUnit)
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java]: Moved this method down to the bottom into the `private class MergeTask` where it is used.

File: hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java [hunk 2]
```diff
@@ -120,93 +122,112 @@
...
-        heartbeatIntervalMillis = heartbeatInterval > 0 ? heartbeatInterval * 1000L : 1000L;
-        pingIntervalMillis = heartbeatIntervalMillis * 10;
+
         node.connectionManager.addConnectionListener(this);
 
         registerMetrics();
     }
 
-    void registerMetrics() {
+    private static long getAsMilliSeconds(GroupProperties.GroupProperty groupProperty) {
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java]: Methods like this maybe a candidate for a Util class, but I didn't pull out any method from this class during this refactoring step.

**Comment by pveentjer** [comment]
I'll give you a tip with such classes. Don't do so many changes in 1 go. Cluster service is one of the most complex classes in the system and totally relies in implicit testing. So what I do, and it is very time consuming, is make a PR with just a few minor changes. But generate a whole sequence of them. This way you can transform the system one step at a time.

**Comment by Donnerbart** [comment]
Yeah, I think none of those `checks=""` classes will be fun to touch and all have a chance to create huge PRs. For classes like this we maybe should do pair-programming with 3 people and merge it right away after the PR builder is green. This may save a lot of time, but of course is difficult in a distributed company like Hazelcast.

**Comment by pveentjer** [comment]
I would really go for a continuous stream of small changes on such classes. Why do you think I have so many commits? :)

**Comment by jerrinot** [comment]
I guess one PR could be ok, but a PR can contain multiple smaller commits - it'll make it easier to review. 
I'll check this one as it is as splitting it up would just increase a risk of a mistake.

**Comment by jerrinot** [comment]
:+1:

**Comment by pveentjer** [comment]
@Donnerbart what are we going to do with this pr? Perhaps @mdogan can have a look at it. My knowledge in the cluster code is not sufficient to review this PR.

**Comment by Donnerbart** [comment]
Knowledge of the cluster code should not be necessary to review this PR. It's about resolving CheckStyle issues, not the logic of the cluster :)

File: hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java (hunk 2 above)
Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java]: :+1:

**Comment by gurbuzali** [comment]
verify

**Comment by gurbuzali** [comment]
will merge once green

**Pull request merged**